src/api/commands/currencyprice/CurrencyPriceRootCommand.ts

```diff
@@ -0,0 +1,36 @@
+import { inject, named } from 'inversify';
+import { RpcRequest } from '../../requests/RpcRequest';
+import { RpcCommandInterface } from '../RpcCommandInterface';
+import { validate, request } from '../../../core/api/Validate';
+import { Logger as LoggerType } from '../../../core/Logger';
+import { Types, Core, Targets } from '../../../constants';
+import { CurrencyPriceAddCommand } from './CurrencyPriceAddCommand';
+import { BaseCommand } from '../BaseCommand';
+import { RpcCommandFactory } from '../../factories/RpcCommandFactory';
+import { Commands } from '../CommandEnumType';
+
+export class CurrencyPriceRootCommand extends BaseCommand implements RpcCommandInterface<void> {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Command) @named(Targets.Command.currencyprice.CurrencyPriceAddCommand) private currencyPriceAddCommand: CurrencyPriceAddCommand,
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
+    ) {
+        super(Commands.CURRENCYPRICE_ROOT);
+        this.log = new Logger(__filename);
+    }
+
+    @validate()
+    public async execute( @request(RpcRequest) data: RpcRequest, rpcCommandFactory: RpcCommandFactory): Promise<any> {
+        return await this.executeNext(data, rpcCommandFactory);
+    }
```

Comment:
move the code from Add command to here. there's no such thing as add-command and theres no subcommands